### PR TITLE
Clean up handling of config

### PR DESCRIPTION
**Pull request by blink1073**
Companion to https://github.com/jupyterlab/jupyterlab/pull/8950.

Moves handling of `page_config` here completely so it is dynamic.

### CODE REVIEW

**Comment by bollwyvl** [comment]
Awesome stuff here!

By dynamic, does this mean config set after starting the server would
update the defaults (with a page reload, presumably)? I'd really love a way
for nbgitpuller to work with this in binder...

**Comment by blink1073** [comment]
> By dynamic, does this mean config set after starting the server would
update the defaults (with a page reload, presumably)?

Yup!

File: jupyterlab_server/config.py [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+    return dynamic_exts
+
+
+def get_static_page_config(app_settings_dir=None, logger=None):
+    # Start with the deprecated `share/jupyter/lab/settings/page_config.json` data
+
+    cm = ConfigManager(config_dir_name="labconfig")
+    page_config = cm.get('page_config')
+
+    # TODO: remove in 4.0
```
Comment (blink1073) [inline jupyterlab_server/config.py:50]: Let's instead allow passing in a `page_config` file here so it is more generic and won't need to change later.  Let's not warn here either.

**Comment by bollwyvl** [comment]
Awesome. Will have to think how this could be made to work the
nbgitpuller...

**Comment by blink1073** [comment]
Released as `jupyterlab_server-2.0.0b7`

**Pull request merged**